test/add_performance_tests.cmake

```diff
@@ -0,0 +1,296 @@
+  endif()
+
+  #
+  # Setup for spinning tests
+  #
+
+  set(SKIP_TEST_SPINNING "${SKIP_TEST}")
+  if(NOT SKIP_TEST_SPINNING)
+    if(PERF_TEST_SKIP_spinning_${TEST_NAME_SYNC} OR
+        PERF_TEST_SKIP_spinning OR NOT ENABLE_SYSTEM_METRIC_COLLECTOR)
+      message(STATUS "Skipping performance tests: spinning_${TEST_NAME_SYNC}")
+      set(SKIP_TEST_SPINNING "SKIP_TEST")
+    endif()
+  endif()
+  set(TEST_ENV_SPINNING ${TEST_ENV})
+  list(APPEND TEST_ENV_SPINNING ${PERF_TEST_ENV_spinning_${TEST_NAME_SYNC}})
+
+  set(NODE_SPINNING_TIMEOUT "30")
+
+  get_filename_component(
+    PERFORMANCE_OVERHEAD_NODE_CSV
+    "${AMENT_TEST_RESULTS_DIR}/${PROJECT_NAME}/overhead_node_test_results_${TEST_NAME_SYNC}.csv"
+    ABSOLUTE
+  )
+  get_filename_component(
+    PERFORMANCE_OVERHEAD_NODE_PNG
+    "${AMENT_TEST_RESULTS_DIR}/${PROJECT_NAME}/overhead_node_test_results_${TEST_NAME_SYNC}.png"
+    ABSOLUTE
+  )
+  list(APPEND TEST_ENV_SPINNING
+    "PERFORMANCE_OVERHEAD_NODE_CSV=${PERFORMANCE_OVERHEAD_NODE_CSV}"
+    "PERFORMANCE_OVERHEAD_NODE_PNG=${PERFORMANCE_OVERHEAD_NODE_PNG}"
+  )
+
+  configure_file(
+    test_spinning.py.in
+    ${CMAKE_CURRENT_BINARY_DIR}/test_spinning_${TEST_NAME_SYNC}.py.configure
+    @ONLY
+  )
+  file(GENERATE
+    OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/test/test_spinning_${TEST_NAME_SYNC}.py"
+    INPUT "${CMAKE_CURRENT_BINARY_DIR}/test_spinning_${TEST_NAME_SYNC}.py.configure"
+  )
+
+  add_launch_test(
+    "${CMAKE_CURRENT_BINARY_DIR}/test/test_spinning_${TEST_NAME_SYNC}.py"
+    TARGET test_spinning_${TEST_NAME_SYNC}
+    ENV ${TEST_ENV_SPINNING}
+    TIMEOUT 120
+    ${SKIP_TEST_SPINNING}
+  )
+  set_tests_properties(test_spinning_${TEST_NAME_SYNC} PROPERTIES
+    RUN_SERIAL TRUE
+    LABELS "performance"
+  )
+
+  #
+  # Setup for cross-vendor tests
+  #
+
+  set(PUBSUB_TIMEOUT "30")
+  set(PERF_TEST_TOPIC ${PERF_TEST_TOPIC_PUB_SUB})
+  get_available_rmw_implementations(_rmw_implementations)
+  foreach(RMW_IMPLEMENTATION_SUB ${_rmw_implementations})
+    # TODO: Why is ${COMM} on the end here? It's always ROS2.
+    set(TEST_NAME_PUB_SUB "${TEST_NAME_SYNC}_${RMW_IMPLEMENTATION_SUB}_${COMM}")
+    set(SKIP_TEST_PUB_SUB "${SKIP_TEST}")
+    if(NOT SKIP_TEST_PUB_SUB)
+      if(PERF_TEST_SKIP_${TEST_NAME_PUB_SUB} OR PERF_TEST_SKIP_pub_sub_${TEST_NAME_SYNC} OR
+          PERF_TEST_SKIP_pub_sub OR PERF_TEST_SKIP_apex_ai_suite OR NOT ENABLE_SYSTEM_METRIC_COLLECTOR)
+        message(STATUS "Skipping performance tests: ${TEST_NAME_PUB_SUB}")
+        set(SKIP_TEST_PUB_SUB "SKIP_TEST")
+      endif()
+    endif()
+    set(TEST_ENV_PUB_SUB ${TEST_ENV})
+    list(APPEND TEST_ENV_PUB_SUB ${PERF_TEST_ENV_pub_sub_${TEST_NAME_SYNC}})
+    list(APPEND TEST_ENV_PUB_SUB ${PERF_TEST_ENV_${TEST_NAME_PUB_SUB}})
+
+    get_filename_component(
+      PERFORMANCE_OVERHEAD_CSV
+      "${AMENT_TEST_RESULTS_DIR}/${PROJECT_NAME}/overhead_test_results_${TEST_NAME_PUB_SUB}.csv"
+      ABSOLUTE
+    )
+    # TODO: Why doesn't this file name have a suffix?
```

Comment:
I think I removed the suffix because I was thinking to join PERFORMANCE_OVERHEAD_PNG and PERFORMANCE_OVERHEAD_CSV in one variable. Then in the Python script we can add the suffix where is needed.

Reply:
I had the same thought, especially now that some of the tests produce more than one file for one variable.
I'd like to do this in a follow-up review. What do you think of `PERF_TEST_RESULT_BASE_PATH`?